cmd/oras/attach.go

```diff
@@ -72,14 +73,23 @@ Example - Attach file 'hi.txt' and add manifest annotations:
 
 Example - Attach file 'hi.txt' and export the pushed manifest to 'manifest.json':
   oras attach --artifact-type doc/example --export-manifest manifest.json localhost:5000/hello:latest hi.txt
+
+Example - Attach file to the manifest tagged 'test' in an OCI layout folder 'layout-dir':
+  oras attach --oci-layout --artifact-type doc/example layout-dir:test hi.txt
 `,
 		Args: cobra.MinimumNArgs(1),
 		PreRunE: func(cmd *cobra.Command, args []string) error {
-			return option.Parse(&opts)
+			opts.RawReference = args[0]
+			opts.FileRefs = args[1:]
+			if err := option.Parse(&opts); err != nil {
+				return err
+			}
+			if opts.Type == option.TargetTypeOCILayout && opts.DistributionSpec.ReferrersAPI != nil && *opts.DistributionSpec.ReferrersAPI {
+				return errors.New("cannot enforce referrers API for image layout target")
```

Comment:
What does it mean if a user uses OCI layout and sets `opts.DistributionSpec.ReferrersAPI` to false? Does it mean they want to use tag schema?

Reply:
Referrers are supported via tag scheme in OCI layout, right? Here we safely ignored the ONLY if the `DistributionSpec.ReferrersAPI` is set to false

Reply:
No, the OCI store does not use tag schema for Referrers.

Reply:
> No, the OCI store does not use tag schema for Referrers.

If so, we should definitely return error.